src/views/account/Homeworks/Homeworks.tsx

```diff
@@ -70,178 +34,644 @@ type HomeworksPageProps = {
 
-  const MemoizedHeaderCalendar = useMemo(
-    () => (
-      <HeaderCalendar
-        epochWeekNumber={epochWeekNumber}
-        oldPageIndex={epochWeekNumber}
-        showPicker={() => {
-          // TODO: Implement date picker logic here
+  const renderWeek = ({ item }) => {
+    const homeworksInWeek = homeworks[item] ?? [];
+
+    const sortedHomework = homeworksInWeek.sort((a, b) => new Date(a.due).getTime() - new Date(b.due).getTime());
+
+    const groupedHomework = sortedHomework.reduce((acc, curr) => {
+      const dayName = getDayName(curr.due);
+      const formattedDate = formatDate(curr.due);
+      const day = `${dayName} ${formattedDate}`;
+
+      if (!acc[day]) {
+        acc[day] = [curr];
+      } else {
+        acc[day].push(curr);
+      }
+
+      // filter homeworks by search terms
+      if (searchTerms.length > 0) {
+        acc[day] = acc[day].filter(homework => {
+          const content = homework.content.toLowerCase().trim().normalize("NFD").replace(/[\u0300-\u036f]/g, "");
+          return content.includes(searchTerms.toLowerCase().trim().normalize("NFD").replace(/[\u0300-\u036f]/g, ""));
+        });
+      }
+
+      // if hideDone is enabled, filter out the done homeworks
+      if (hideDone) {
+        acc[day] = acc[day].filter(homework => !homework.done);
+      }
+
+      // remove all empty days
+      if (acc[day].length === 0) {
+        delete acc[day];
+      }
+
+      return acc;
+    }, {} as Record<string, Homework[]>);
+
+    return (
+      <ScrollView
+        style={{ width: finalWidth, height: "100%"}}
+        contentContainerStyle={{
+          padding: 16,
+          paddingTop: insets.top + 56,
         }}
-        changeIndex={(index: number) => manuallyChangeWeek(index)}
-      />
-    ),
-    [epochWeekNumber, manuallyChangeWeek]
-  );
+        showsVerticalScrollIndicator={false}
+        refreshControl={
+          <RefreshControl
+            refreshing={refreshing}
+            onRefresh={() => updateHomeworks(true)}
+            progressViewOffset={insets.top + 56}
+          />
+        }
+      >
+        {groupedHomework && Object.keys(groupedHomework).map((day, index) => (
+          <Reanimated.View
+            key={day}
+            entering={animPapillon(FadeInUp)}
+            exiting={animPapillon(FadeOutDown)}
+            layout={animPapillon(LinearTransition)}
+          >
+            <NativeListHeader animated label={day} />
 
-  useLayoutEffect(() => {
-    navigation.setOptions({
-      headerTitle: () => MemoizedHeaderCalendar,
-    });
-  }, [navigation, epochWeekNumber]);
+            <NativeList animated>
+              {groupedHomework[day].map((homework, idx) => (
+                <HomeworkItem
+                  key={homework.id}
+                  index={idx}
+                  total={groupedHomework[day].length}
+                  homework={homework}
+                  onDonePressHandler={async () => {
```

Comment:
POur add ajouter un truc pour mieux voir les devoirs qui sont recherchés
```suggestion
                  searchTerms={searchTerms}
```